lib/Driver/ToolChain.cpp

```diff
@@ -118,11 +118,23 @@ ToolChain::constructJob(const JobAction &JA,
     }
   }
 
+  const char *responseFilePath = nullptr;
+  llvm::opt::ArgStringList responseFileArg = {};
+  if (!llvm::sys::commandLineFitsWithinSystemLimits(executablePath,
+                                                    invocationInfo.Arguments)
+      && JA.allowsResponseFiles()) {
```

Comment:
Nitpick: You should probably test these two in the opposite order, so we don't waste time measuring arguments if we can't use a response file anyway.